xilem/examples/stopwatch.rs

```diff
@@ -0,0 +1,260 @@
+// Copyright 2024 the Xilem Authors
+// SPDX-License-Identifier: Apache-2.0
+
+//! A stopwatch to display elapsed time.
+
+use masonry::dpi::LogicalSize;
+use masonry::event_loop_runner::{EventLoop, EventLoopBuilder};
+use masonry::widget::{Axis, CrossAxisAlignment, MainAxisAlignment};
+use std::ops::{Add, Sub};
+use std::time::{Duration, SystemTime};
+use tokio::time;
+use tracing::warn;
+use winit::error::EventLoopError;
+use winit::window::Window;
+use xilem::view::{async_repeat, button, flex, label, AnyFlexChild, FlexExt, FlexSpacer};
+use xilem::{WidgetView, Xilem};
+use xilem_core::fork;
+use xilem_core::one_of::Either;
+
+/// The state of the entire application.
+///
+/// This is owned by Xilem, used to construct the view tree, and updated by event handlers.
+struct Stopwatch {
+    active: bool,
+    // The duration to add to the duration since the last instant.
+    // This is needed since you can pause a timer, and we need to account for all
+    // time the timer was active before the last start.
+    added_duration: Duration,
+    // The absolute time of the last start for calculating elapsed time.
+    last_start_time: Option<SystemTime>,
+    // The duration displayed; updated by by `update_display()`
+    displayed_duration: Duration,
+    // An error string to display if there is an error.
+    displayed_error: String,
+    // A list of the length of all completed splits. Does not include the current split.
+    completed_lap_splits: Vec<Duration>,
+    // The duration of the main timer when the split was started.
+    split_start_time: Duration,
```

Comment:
I guess these could also be doc-comments, for better documentation support in editors?

Reply:
Done.